src/client/array/dc_array.c

```diff
@@ -1929,34 +1930,33 @@ dc_array_get_size(tse_task_t *task)
 	query_args->recx	= &kqp->recx;
 	query_args->max_epoch	= NULL;
 
-	rc = tse_task_register_comp_cb(query_task, get_array_size_cb, &kqp,
-				       sizeof(kqp));
-	if (rc != 0)
-		D_GOTO(err_query_task, rc);
-
-	rc = tse_task_register_deps(task, 1, &query_task);
+	rc = tse_task_register_comp_cb(query_task, get_array_size_cb, &kqp, sizeof(kqp));
 	if (rc != 0)
-		D_GOTO(err_query_task, rc);
+		D_GOTO(err_task, rc);
 
 	rc = tse_task_register_comp_cb(task, free_query_cb, &kqp, sizeof(kqp));
 	if (rc != 0)
-		D_GOTO(err_query_task, rc);
-
-	rc = tse_task_schedule(query_task, false);
-	if (rc != 0)
-		D_GOTO(err_query_task, rc);
+		D_GOTO(err_task, rc);
 
-	tse_sched_progress(tse_task2sched(task));
```

Comment:
why remove this progress function?

Reply:
My understand is that the array API daos_array_get_size() will handle the process via dc_task_schedule(). That is the common logic for all the others APIs, for example the ones in src/client/api/object.c

Reply:
OK i missed the part where you changed tse_task_schedule to be called with instant = true. so yes that is equivalent, although different from what is being done in other dc_array calls.